File: packages/session-service/internal/service/speaker.go
package service

type Speaker struct {
	ID         string    `json:"id" gorm:"type:uuid"`
	Name       string    `json:"name" gorm:"not null"`
	Bio        string    `json:"bio" gorm:"not null"`
	Headline   string    `json:"headline" gorm:"not null"`
	Photo      string    `json:"photo" gorm:"not null"`
	Sessions   []Session `json:"sessions,omitempty" gorm:"many2many:session_speakers;constraint:OnDelete:CASCADE;"`
}

File: packages/speaker-service/src/entity/Speaker.ts
import {
  BaseEntity,
  Entity,
  PrimaryGeneratedColumn,
  ManyToMany,
  OneToMany,
  JoinTable,
  Column,
} from "typeorm";
import Session from "./Session";

@Entity("speakers")
export default class Speaker extends BaseEntity {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column()
  name!: string;

  @Column()
  bio!: string;

  @Column()
  photo!: string;

  @Column()
  headline!: string;

  @ManyToMany(() => Session, (session) => session.speakers)
  @JoinTable({
    name: "session_speakers",
    joinColumn: {
      name: "speaker_id",
      referencedColumnName: "id",
    },
    inverseJoinColumn: {
      name: "session_id",
      referencedColumnName: "id",
    },
  })
  sessions!: Session[];
}
